spec/unit/provider/package/yum_spec.rb

```diff
@@ -248,6 +248,93 @@
       @provider = Chef::Provider::Package::Yum.new(@new_resource, @run_context)
       @provider.load_current_resource
       expect(@new_resource.package_name).to eq("test-package")
+      expect(@new_resource.version).to eq(nil)
+    end
+
+    it "should search provides if a package name can't be found then set version to match if a requirement was passed in the package name" do
```

Comment:
It looks like there's a lot of duplication in these three tests. Start with moving @yum_cache to a let and put these in a context?

```
context "when the package name isn't found" do
  let(:yum_cache) { double(
          'Chef::Provider::Yum::YumCache',
          :reload_installed => true,
          :reset => true,
          :installed_version => "1.0.1.el5",
          :candidate_version => "2.0.1.el5",
          :package_available? => false,
          :version_available? => true,
          :disable_extra_repo_control => true
    )
  }

  before do
     allow(Chef::Provider::Package::Yum::YumCache).to receive(:instance).and_return(yum_cache)
     pkg = Chef::Provider::Package::Yum::RPMPackage.new("test-package", "2.0.1.el5", "x86_64", [])
     expect(yum_cache).to receive(:packages_from_require).and_return([pkg])
  end

  it "searches provides and sets version to match if a version constraint was passed in the version" do
    # ...
  end

  # ...
end
```